src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java

```diff
@@ -159,9 +159,41 @@ public Optional<VoltageControl<?>> getVoltageControl(VoltageControl.Type type) {
         return getVoltageControls().stream().filter(vc -> vc.getType() == type).findAny();
     }
 
-    @Override
-    public Optional<VoltageControl<?>> getHighestPriorityMainVoltageControl() {
-        return VoltageControl.findMainVoltageControlsSortedByPriority(this).stream().findFirst();
+    private int firstPositionInList(String str, List<String> lStr) {
```

Comment:
I think you must add an enum, for parameters and this kind of method that we don't want to see.

Reply:
It should be better now.